cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java

```diff
@@ -104,7 +104,7 @@ private void upgradeSystemDatasets() throws Exception {
 
   private void upgradeUserTables() throws Exception {
     for (final NamespaceMeta namespaceMeta : namespaceQueryAdmin.list()) {
-      impersonator.doAs(namespaceMeta, new Callable<Void>() {
+      impersonator.doAs(namespaceMeta.getNamespaceId(), new Callable<Void>() {
```

Comment:
No, it will not. Upgrader is in my list of items to fix. I am not sure how to get owners for each table but I will work on that after the RC. Will also open a JIRA.